### PR TITLE
frontend: Add CNCF and verification badge to charts

### PR DESCRIPTION
**Pull request by vyncent-t**
# Enhance App Catalog with ArtifactHub Badges

## Description

fixes issue #20

This PR aims to enhance our app catalog by incorporating important badges from ArtifactHub. These badges indicate whether an app is official, verified, or a CNCF project. Displaying these badges will help users make informed decisions about which packages to install, ensuring they opt for secure and verified applications.

## Changes

- [x] Added support for displaying ArtifactHub badges (official, verified, and CNCF project) in the app catalog.
- [x] Updated the UI to prominently feature these badges next to corresponding apps in the catalog.
- [x] Ensured that the CNCF badge is included and correctly displayed for relevant apps.
- [x] Implemented logic to fetch and display badge information from ArtifactHub.

## Verification

- [x] Verified that the badges are correctly displayed for apps that are marked as official, verified, or CNCF projects.
- [x] Tested the UI to ensure that the addition of badges does not disrupt the existing layout or functionality of the app catalog.
- [x] Confirmed that the badges provide clear and accurate information to users about the status and security of the apps.

## Images
image

### CODE REVIEW

File: app-catalog/src/components/charts/List.tsx [hunk 1]
```diff
@@ -158,25 +159,48 @@ export function ChartsList({ fetchCharts = fetchChartsFromArtifact }) {
         ) : charts.length === 0 ? (
           <Box mt={2} mx={2}>
             <Typography variant="h5" component="h2">
-              {`No charts found for ${
-                search ? `search term: ${search}` : `category: ${chartCategory.title}`
-              }`}
+              {`No charts found for ${search ? `search term: ${search}` : `category: ${chartCategory.title}`}`}
```
Comment (joaquimrocha) [inline app-catalog/src/components/charts/List.tsx]: Maybe this was not supposed to be changed?

File: app-catalog/src/components/charts/List.tsx [hunk 2]
```diff
@@ -158,25 +159,48 @@ export function ChartsList({ fetchCharts = fetchChartsFromArtifact }) {
...
                         margin: '1rem',
+                        alignSelf: 'flex-start',
                       }}
                       component="img"
                     />
+                    <Box display="flex" alignItems="center" justifyContent="space-around" marginRight="10px" >
+                      {chart.repository.cncf && (
+                        <Tooltip title="CNCF Project">
+                          <CardMedia
+                            image={`https://github.com/cncf/artwork/blob/1e04eb8355adcf8ffd7bb40ff1da098ea80452e3/other/cncf/icon/color/cncf-icon-color.png?raw=true`}
```
Comment (illume) [inline app-catalog/src/components/charts/List.tsx]: Please bring this icon into the plugin.

Comment (joaquimrocha) [inline app-catalog/src/components/charts/List.tsx]: Please download the icon and use it directly. You shouldn't need a CardMedia for this.

File: app-catalog/src/components/charts/List.tsx [hunk 3]
```diff
@@ -158,25 +159,48 @@ export function ChartsList({ fetchCharts = fetchChartsFromArtifact }) {
...
+                              width: '30px',
+                              alignSelf: 'flex-end',
+                            }}
+                            component="img"
+                          />
+                        </Tooltip>
+                      )}
+                      {chart.official && (
+                        <Tooltip title="Official">
+                          <Icon icon="mdi:shield-check" style={{ fontSize: '2em' }} />
```
Comment (joaquimrocha) [inline app-catalog/src/components/charts/List.tsx]: Please use the [star](https://icon-sets.iconify.design/mdi/star-circle/) icon for this.

File: app-catalog/src/components/charts/List.tsx [hunk 4]
```diff
@@ -158,25 +159,48 @@ export function ChartsList({ fetchCharts = fetchChartsFromArtifact }) {
...
                       }}
                       component="img"
                     />
+                    <Box display="flex" alignItems="center" justifyContent="space-around" marginRight="10px" >
+                      {chart.repository.cncf && (
+                        <Tooltip title="CNCF Project">
+                          <CardMedia
+                            image={`https://github.com/cncf/artwork/blob/1e04eb8355adcf8ffd7bb40ff1da098ea80452e3/other/cncf/icon/color/cncf-icon-color.png?raw=true`}
+                            style={{
+                              width: '30px',
```
Comment (joaquimrocha) [inline app-catalog/src/components/charts/List.tsx]: Please make the other icons 22px and try to make this one match that size.

**Comment by illume** [comment]
@vyncent-t from our discussions...

We were looking at showing the "not verified states" too.
image

Notice the extra description, explaining what exactly each of the badges means https://artifacthub.io/packages/helm/nginx-gateway-fabric/nginx-gateway-fabric

Additionally, perhaps sharing the iconography as artifact hub will make it easier for users. So they can reuse that knowledge of what icon is what?

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
Left a couple of notes. It would also be nice if we sorted the list of charts to show them in this order:
- CNCF
- Official
- Verified
- rest

File: app-catalog/src/components/charts/List.tsx (hunk 4 above)
Comment (vyncent-t) [reply app-catalog/src/components/charts/List.tsx]: fixed!

File: app-catalog/src/components/charts/List.tsx (hunk 3 above)
Comment (vyncent-t) [reply app-catalog/src/components/charts/List.tsx]: done!

File: app-catalog/src/components/charts/List.tsx (hunk 2 above)
Comment (vyncent-t) [reply app-catalog/src/components/charts/List.tsx]: done!

File: app-catalog/src/components/charts/List.tsx (hunk 1 above)
Comment (vyncent-t) [reply app-catalog/src/components/charts/List.tsx]: done!

**Comment by vyncent-t** [comment]
I updated the visuals to match those of artifact hub with matching labels for non verified and others @illume 

image

**Comment by sniok** [comment]
Is it just me or does it look a bit too "busy"? A lot of different colors makes it a bit hard to parse visually. What if we kept the icons only when the app is "verified", "official", etc and kept them monochrome at all times? Something like this:

image

doesn't have to be changed within this PR, we can discuss and iterate later

**Comment by vyncent-t** [comment]
noticed in some places artifact hub will use either chart.cncf or chart.repository.cncf on their icons to choose to display so I implemented that

**Comment by joaquimrocha** [comment]
I have reduced the size of the cncf icon slightly + made the official badge appear first, as we had chatted (most important to least important).
